ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/evpn-fabric-super-spine-yml.j2

```diff
@@ -8,6 +8,78 @@
+{%                 if some_switch_vars.type == 'l3leaf' %}
+{%                     for l3leaf_node_group in some_switch_vars.l3leaf.node_groups | arista.avd.natural_sort %}
+{%                         for node in some_switch_vars.l3leaf.node_groups[l3leaf_node_group].nodes | arista.avd.natural_sort %}
+{%                             if node == some_switch %}
+{%                                 set peer.id = some_switch_vars.l3leaf.node_groups[l3leaf_node_group].nodes[node].id %}
+{%                                 if some_switch_vars.l3leaf.node_groups[l3leaf_node_group].bgp_as is defined %}
+{%                                     set peer.bgp_as = some_switch_vars.l3leaf.node_groups[l3leaf_node_group].bgp_as %}
+{%                                 else %}
+{%                                     set peer.bgp_as = some_switch_vars.l3leaf.defaults.bgp_as %}
+{%                                 endif %}
+{%                             endif %}
+{%                         endfor %}
+{%                     endfor %}
+{%                     set peer.ip = some_switch_vars.overlay_loopback_network_summary | ipaddr('network') | ipmath(peer.id + some_switch_vars.spine.nodes | length) %}
+{%                 elif some_switch_vars.type == 'spine' %}
+{%                     set peer.bgp_as = some_switch_vars.spine.bgp_as %}
+{%                     set peer.ip = some_switch_vars.overlay_loopback_network_summary | ipaddr('network') | ipmath(some_switch_vars.spine.nodes[some_switch].id) %}
+{%                 elif some_switch_vars.type == 'super-spine' %}
+{%                     set peer.bgp_as = some_switch_vars.super_spine.bgp_as %}
+{%                     set peer.ip = some_switch_vars.super_spine_loopback_network_summary  | ipaddr('network') | ipmath(some_switch_vars.super_spine.nodes[some_switch].id) %}
+{%                 elif some_switch_vars.type == 'overlay-controller' %}
+{%                     if some_switch_vars.overlay_controller.nodes[some_switch].bgp_as is defined %}
+{%                         set peer.bgp_as = some_switch_vars.overlay_controller.nodes[some_switch].bgp_as %}
+{%                     else %}
+{%                         set peer.bgp_as = some_switch_vars.overlay_controller.defaults.bgp_as %}
+{%                     endif %}
+{%                     set peer.ip = some_switch_vars.overlay_controller_loopback_network_summary | ipaddr('network') | ipmath(some_switch_vars.overlay_controller.nodes[some_switch].id) %}
+{%                 endif %}
+{%                 do switch.evpn_overlay_peers.append(peer) %}
+{%             endif %}
+{%         endfor %}
+{%     endif %}
+{% endfor %}
+{# Look for evpn_overlay_controllers / servers where we are client #}
+{% if evpn_overlay_controller_groups is defined and evpn_overlay_controller_groups is not none %}
+{%     for evpn_overlay_controller_group in evpn_overlay_controller_groups %}
+{%         for some_switch in groups[evpn_overlay_controller_group] %}
```

Comment:
Maybe we should use a more specific name to relate `some_switch` ? Like `evpn_overlay_switch` as it is only iteration over `groups[evpn_overlay_controller_group]`